tests/components/ssdp/test_init.py

```diff
@@ -1,143 +1,130 @@
 """Test the SSDP integration."""
-import asyncio
-from datetime import timedelta
+from datetime import datetime, timedelta
 from ipaddress import IPv4Address, IPv6Address
-from unittest.mock import patch
+from unittest.mock import ANY, AsyncMock, patch
 
-import aiohttp
-from async_upnp_client.search import SSDPListener
+from async_upnp_client.ssdp import udn_from_headers
+from async_upnp_client.ssdp_listener import SsdpListener
 from async_upnp_client.utils import CaseInsensitiveDict
 import pytest
 
+import homeassistant
 from homeassistant import config_entries
 from homeassistant.components import ssdp
 from homeassistant.const import (
     EVENT_HOMEASSISTANT_STARTED,
     EVENT_HOMEASSISTANT_STOP,
     MATCH_ALL,
 )
-from homeassistant.core import CoreState, callback
 from homeassistant.setup import async_setup_component
 import homeassistant.util.dt as dt_util
 
-from tests.common import async_fire_time_changed, mock_coro
+from tests.common import async_fire_time_changed
 
 
-def _patched_ssdp_listener(info, *args, **kwargs):
-    listener = SSDPListener(*args, **kwargs)
-
-    async def _async_callback(*_):
-        await listener.async_callback(info)
-
-    @callback
-    def _async_search(*_):
-        # Prevent an actual scan.
-        pass
-
-    listener.async_start = _async_callback
-    listener.async_search = _async_search
-    return listener
-
-
-async def _async_run_mocked_scan(hass, mock_ssdp_response, mock_get_ssdp):
-    def _generate_fake_ssdp_listener(*args, **kwargs):
-        return _patched_ssdp_listener(
-            mock_ssdp_response,
-            *args,
-            **kwargs,
-        )
+def _ssdp_headers(headers):
+    return CaseInsensitiveDict(
+        headers, _timestamp=datetime(2021, 1, 1, 12, 00), _udn=udn_from_headers(headers)
+    )
 
-    with patch(
-        "homeassistant.components.ssdp.async_get_ssdp",
-        return_value=mock_get_ssdp,
-    ), patch(
-        "homeassistant.components.ssdp.SSDPListener",
-        new=_generate_fake_ssdp_listener,
-    ), patch.object(
-        hass.config_entries.flow, "async_init", return_value=mock_coro()
-    ) as mock_init:
-        assert await async_setup_component(hass, ssdp.DOMAIN, {ssdp.DOMAIN: {}})
-        hass.bus.async_fire(EVENT_HOMEASSISTANT_STARTED)
-        await hass.async_block_till_done()
-        await hass.async_block_till_done()
 
-    return mock_init
+async def init_ssdp_component(hass: homeassistant) -> SsdpListener:
```

Comment:
Maybe we should make this a common test helper that can mock an ssdp search response, so that we at least extract the details to a single place?